### DISCUSSION TITLE
SMARTS returning a RunReactants product that cannot be kekulised

### DISCUSSION THREAD

**Discussion by hejops** [rdkit/rdkit#4083] [Q&A] [answered]
I want to use `RunReactants` to get the SMILES of an aromatic product (containing a positive charge). I can get the non-kekulised form just fine, but attempting to kekulise always throws an error. Does anyone know how my SMARTS can be improved?

```
from rdkit import Chem
from rdkit.Chem import rdChemReactions

smarts = "[OH2:1].[cH:11][c:12][c:13][o+1:14]>>[OH2+:1][c:11][c:12][c:13][o+0:14]"

reac1 = "O"
reac2 = "C1=CC2=CC3=CC=CC=C3[O+]=C2C=C1"
reactant_tuple = (Chem.MolFromSmiles(reac1), Chem.MolFromSmiles(reac2))
reaction = rdChemReactions.ReactionFromSmarts(smarts, useSmiles=False)

product = reaction.RunReactants(reactant_tuple)[0][0]

print(Chem.MolToSmiles(product, kekuleSmiles=False))
# "[OH2+]c1cccc2oc3ccccc3cc12"

Chem.SanitizeMol(
    product,
    Chem.SanitizeFlags.SANITIZE_FINDRADICALS
    | Chem.SanitizeFlags.SANITIZE_KEKULIZE
    | Chem.SanitizeFlags.SANITIZE_SETAROMATICITY
    | Chem.SanitizeFlags.SANITIZE_SETCONJUGATION
    | Chem.SanitizeFlags.SANITIZE_SETHYBRIDIZATION
    | Chem.SanitizeFlags.SANITIZE_SYMMRINGS,
)
print(Chem.MolToSmiles(product, kekuleSmiles=True))
# desired:
# "C1=CC2C([OH2+])C3=CC=CC=C3OC=2C=C1"
# aromaticity and positive charge should be preserved
# (the position of -OH2+ doesn't really matter)
```

**Comment by ptosco** [comment, marked as answer]
@hejops What reaction are you attempting to perform here? As it is written, that looks like the nucleophilic addition of a water molecule on an aromatic system (which is a bit odd), but with an incorrect hydrogen count in the SMARTS:
![image](https://user-images.githubusercontent.com/5244385/115986969-3f05b600-a5b3-11eb-8e65-dca37e42617e.png)
The resulting product should still have a hydrogen attached at the same position where the water nucleophilic attack has taken place, while your reaction SMARTS indicates that the hydrogen should leave, which seems wrong.
Anyway, the resulting system is not aromatic anymore, as expected from a nucleophilic addition on an aromatic system, hence it fails to kekulize.

**Reply by hejops** [reply]
"nucleophilic addition of a water molecule on an aromatic system" would be correct, more or less. (I have a list of reactants to be merged according to some relatively simple SMARTS, and this is one of my edge cases)

Retaining the H on C11 successfully returned the (kekulised) product that you drew. Indeed, I should have noticed that the H would remain, resulting in a loss of aromaticity. Thanks for pointing it out.
